### PR TITLE
Refactor : Vitest to EventVolunteers/VolunteerGroups Screen

### LINKED ISSUE #2549: Refactor: `src/screens/EventVolunteers/VolunteerGroups` from Jest to Vitest
**Is your feature request related to a problem? Please describe.**
This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor this file's test cases to align with Vitest's syntax and features.

**Acceptance Criteria**
1. Replace Jest-specific functions and mocks with Vitest equivalents
2. Ensure all tests in `src/screens/EventVolunteers/VolunteerGroups` pass after migration using `npm run test:vitest`
3. Maintain the test coverage for the file as 100% after migration
4. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation:
https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by shivasankaran18**
**What kind of change does this PR introduce?**

Added Vitest to EventVolunteers/VolunteerGroups screen

**Issue Number: 2549**

Fixes #2549  

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**
Screenshot 2024-12-21 103046

**If relevant, did you update the documentation?**

**Summary**

Migrated the testing framework to Vitest.
Updated all test files and configurations to be compatible with Vitest's syntax and features.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **Tests**
	- Enhanced test flexibility by updating the mocking strategy for `useParams` in the `VolunteerGroups` component.
	- Improved error handling and scenario coverage in tests, including cases for undefined and valid parameters.
	- Transitioned from Jest to Vitest for mocking methods, ensuring better compatibility and reliability.

### CODE REVIEW

**Comment by shivasankaran18** [comment]
@coderabbit review..

**Comment by palisadoes** [comment]
1. Please make sure that CodeRabbit.ai approves the PR.

**Comment by palisadoes** [comment]
1. Ensure that CodeRabbit.ai approves your work

File: src/screens/EventVolunteers/VolunteerGroups/VolunteerGroups.spec.tsx [hunk 1]
```diff
@@ -61,19 +62,26 @@ const renderVolunteerGroups = (link: ApolloLink): RenderResult => {
...
   });
 
   afterAll(() => {
-    jest.clearAllMocks();
+    vi.clearAllMocks();
   });
 
   it('should redirect to fallback URL if URL params are undefined', async () => {
+    /**  Mocking the useParams hook to return undefined parameters */
+    vi.mocked(useParams).mockReturnValue({ orgId: '', eventId: '' });
```
[the remark this answers is not in the document]
Comment (shivasankaran18) [reply src/screens/EventVolunteers/VolunteerGroups/VolunteerGroups.spec.tsx]: Resolved

**Comment by shivasankaran18** [comment]
@palisadoes  done with the changes

**Comment by shivasankaran18** [comment]
@coderabbit review

**Comment by shivasankaran18** [comment]
@palisadoes @dhanagopu I have merged 4 to 5 PRs for covering jest to vitest tests..Can I create a single issue of converting all the screens and components tests to vitest ..I would like to work on it . Could I have your suggestion?

**Comment by palisadoes** [comment]
1. Please make sure that coderabbit.ai approves the PR.
2. Make the appropriate changes as suggested.

File: src/screens/EventVolunteers/VolunteerGroups/VolunteerGroups.spec.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (shivasankaran18) [reply src/screens/EventVolunteers/VolunteerGroups/VolunteerGroups.spec.tsx]: could you approve my PR now

**Comment by shivasankaran18** [comment]
@coderabbit done with your changes..could you review and approve my PR

**Comment by shivasankaran18** [comment]
@palisadoes  @dhanagopu done with the changes..I have merged 4 to 5 PRs for covering jest to vitest tests..Can I create a single issue of converting all the screens and components tests to vitest ..I would like to work on it . Could I have your suggestion?

**Comment by palisadoes** [comment]
> @palisadoes @dhanagopu done with the changes..I have merged 4 to 5 PRs for covering jest to vitest tests..Can I create a single issue of converting all the screens and components tests to vitest ..I would like to work on it . Could I have your suggestion?

Thanks, we are dividing the work into multiple issues so that everyone can get a chance.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**